### DISCUSSION TITLE
-inverse-search appears to be deprecated. Any alternatives?

### DISCUSSION THREAD

**Discussion by ubaldot** [sumatrapdfreader/sumatrapdf#4607] [Q&A]
I am in need of  setting the backwards search through command line but reading at the docs this flag seems to be deprecated. Any alternatives?

**Comment by GitHubRulesOK** [comment]
-inverse-search is a ONE OFF requirement at installation thus not needed on a daily basis

The problem is down to the web will keep more copies of disinformation than good information!

When you set -inverse search is down to users configuration so for a Portable LaTeX or first time NOOB  user it may be worth including as a command but for an experienced user never call it.

By default for SumatraPDF it is OFF and not initiated thus the advice to initiate it via the "editor" -inverse-search ***first run*** thus I can trigger 
SumatraPDF -inverse-search "Notepad.exe ""%f"" %l" and actually see the fail so better by hand in options or in Advanced settings
![image](https://github.com/user-attachments/assets/46a43aea-3801-42e8-83c5-4a745999bce8)
![image](https://github.com/user-attachments/assets/ec139a60-359f-4f4c-9f88-5a83dffa0acb)
![image](https://github.com/user-attachments/assets/5e59fd35-3258-47eb-a7f1-32e5577dffe8)

**Comment by ubaldot** [comment]
Ok.
I am writing a Vim plugin that provides a command that automatically does the following

1. Build the pdf
2. Open SumatraPDF 

At this point I want the user to be able to forward/backwards search provided that Sumatra options are correctly set. 

At the moment, I am asking the user to manually copy paste the line `gvim --servername gvim --remote-send ":call BackwardSearch(%l, '%f')<cr>"`  in Sumatra options as a setup step. Not a big issue but I would avoid asking this. I would prefer to include this in the step 2. above, so the user does not have to bother to do manual setup. By looking at the docs, I found  `-inverse-search` but it seems deprecated.

**Reply by GitHubRulesOK** [reply]
simply ask them to paste the correct line into settings ONCE and backup the settings once proven as working so it is NOT disturbed on every single "view a pdf"

e.g. where is gvim ? is it an exe ? and Windows quotes for spaces are **"normally double"**
```
EnableTeXEnhancements = true
InverseSearchCmdLine = "C:\windows user path\gvim.exe" parameters etc. .........%l ...."%f"
```

**Reply by ubaldot** [reply]
Yeah, that is what I am actually doing (and it works). I was just wondering if I could avoid to ask users such additional step but I wanted to provide a more "out-of-the-box" experience. That was all.

**Reply by GitHubRulesOK** [reply]
the problem is if you try to automate it as many do it usually breaks more times than works. 

Teaching a person to phish for themselves, is best support in the longer term.

**Reply by GitHubRulesOK** [reply]
I think we are still waiting for your edits to be added into the website & application docs as they were acceptable but not yet fully committed ! due to current release freeze?

**Reply by GitHubRulesOK** [reply]
There is a question as to what settings are best for gvim since its inclusion in 2009 it has clearly altered 

https://github.com/sumatrapdfreader/sumatrapdf/commit/2a5790078010a67e2e06b199f9834aa72161781d#diff-54416cdfdc1ce587ba37065f57ecd3b550b09629980a9afc041cd368dabde36bR638

current code has it entered as differently twice ! https://github.com/sumatrapdfreader/sumatrapdf/blob/415988f600ca3dfee2836ac7c35e025d3b6b9fcd/src/AppTools.cpp#L285
https://github.com/sumatrapdfreader/sumatrapdf/blob/415988f600ca3dfee2836ac7c35e025d3b6b9fcd/src/AppTools.cpp#L294

So it would be best if there were no auto detect by EITHER app as time changes desired functions. and I dont know why this issue is open ! https://github.com/sumatrapdfreader/sumatrapdf/issues/1140


there should only be one current gvim version method but we now see variously

gvim --remote-silent +$lineNumber "$fileFullName" # the line to invoke inverse search  
"gvim.exe", "%f" +%l  
"gvim.exe", -c ":RemoteOpen +%l %f"  

which should it be ?
is down to GVIM guidance

**Reply by ubaldot** [reply]
There are infinite gvim settings: `https://vimhelp.org/remote.txt.html`. The one I use works very well and it is very flexible. But regardless, the only important thing is to pass the arguments  %l and %f .whatever request you send to the gvim server. So there is not ONE possible settings, but infinite.

**Reply by GitHubRulesOK** [reply]
One gvim replacement was supposedly neovim

Thus for example the last MINIMAL UNIVERSAL windows 32 bit + / Windows 7 + NVIM v0.5.1 compatible call was
```
"path to neovim\bin\nvim.exe"  +%l -- "%f" 
```
gvim or vim should still be similar MINIMAL using the ethos of `--` separates commands from filenames in ALL filenames methods
```
"path to gvim\vim91\gvim.exe"  +%l -- "%f" 
```

Just tested both those work well with gvim and neovim so any other enhancement beyond those are extra to SumatraPDF requirements.

**SumatraPDF only NEEDS** where # is nothing or `n` or `g`
```
EnableTeXEnhancements = true
InverseSearchCmdLine = "drive:\blah blah\#vim.cmd"  +%l -- "%f"
```

However I have problems unless that is a CMD pointing to an exe as the exe will not behave as expected

![image](https://github.com/user-attachments/assets/4018d470-9401-42d2-bb90-cc81d4fb12a8)

**Reply by ubaldot** [reply]
I don't understand what you want to say tbh... For me the discussion is closed. I have a working solution, I only asked if it was possible to inline a setting and the answer was no. Fair enough. Not sure now what this gvim specific discussion is about.

**Reply by GitHubRulesOK** [reply]
The answer is the user should define their own desired variables
but for a generic app such as vim it should be one setting once
that can be done during installation using sumatrapdf -install -inverse-search "path to vim.cmd"  +%l -- "%f"
